stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/OnlyFileAttachmentsViewHolder.kt

```diff
@@ -56,5 +70,60 @@ internal class OnlyFileAttachmentsViewHolder(
         super.bindData(data, diff)
 
         binding.fileAttachmentsView.setAttachments(data.message.attachments)
+
+        val uploadIdList: List<String> = data.message
+            .attachments
+            .filter { attachment -> attachment.uploadState == Attachment.UploadState.InProgress }
+            .mapNotNull { attachment -> attachment.uploadId }
+
+        val needUpload = uploadIdList.isNotEmpty()
+
+        if (needUpload) {
+            clearScope()
+            val scope = CoroutineScope(DispatcherProvider.Main)
+            this.scope = scope
+
+            scope.launch {
+                trackFilesSent(uploadIdList)
+            }
+        } else {
+            binding.sentFiles.isVisible = false
+        }
+    }
+
+    private suspend fun trackFilesSent(uploadIdList: List<String>) {
+        var filesSent = 0
+        val totalFiles = uploadIdList.size
+
+        binding.sentFiles.isVisible = true
+        binding.sentFiles.text =
+            String.format(context.getString(R.string.stream_ui_upload_sending), filesSent, totalFiles)
+
+        uploadIdList.map { uploadId ->
+            ProgressTrackerFactory.getOrCreate(uploadId).isComplete().map { listOf(it) }
+        }.reduce { acc, flow ->
+            acc.combine(flow) { flow1, flow2 -> flow1 + flow2 }
+        }.map { stateList ->
+            stateList.filter { it }
+        }.collect { stateList ->
+            filesSent = stateList.size
+
+            if (filesSent == totalFiles) {
+                binding.sentFiles.text = context.getString(R.string.stream_ui_upload_complete)
+            } else {
+                binding.sentFiles.text =
+                    String.format(context.getString(R.string.stream_ui_upload_sending), filesSent, totalFiles)
+            }
+        }
```

Comment:
I think this is the same logic - but please do check

```
val completionFlows: List<Flow<Boolean>> = uploadIdList.map { uploadId ->
    ProgressTrackerFactory.getOrCreate(uploadId).isComplete()
}
combine(completionFlows) { values -> values.count { complete -> complete } }
    .collect { completedCount ->
        filesSent = completedCount
        // ...
    }
```

Reply:
Much simpler o.O. I didn't know about the `combine` operator